### PR TITLE
Improve build time by getting rid of the sdk dependencies!

### PR DESCRIPTION
**Pull request by shareefalis**
- fix async  dependencies.sh to make it more robust 
- Skip getting submodules and  most of the dependencies  if  we have the cached SDK ready to go

### CODE REVIEW

File: Tools/dependencies.sh [hunk 1]
```diff
@@ -3,21 +3,17 @@ set -xe
...
 wget --no-check-certificate https://raw.githubusercontent.com/FFmpeg/gas-preprocessor/master/gas-preprocessor.pl && chmod +x gas-preprocessor.pl && sudo mv -f gas-preprocessor.pl /usr/local/bin &
 bundle install &
 
-cat .gitmodules | grep -o '".*"' | cut -d '"' -f 2 | while sleep 2 && read line; do git submodule update --init --recursive $line & done
-# make sure everything is fetched
 
-brew update 1>/dev/null && brew install doxygen nasm yasm optipng imagemagick intltool ninja antlr cmake 
+# make sure everything is fetched
 
-[ -x /usr/local/bin/libtoolize ] || sudo ln -sf /usr/local/bin/glibtoolize /usr/local/bin/libtoolize
```
Comment (ianblenke) [inline Tools/dependencies.sh:18]: Why did you remove this libtoolize symlink?

File: Tools/dependencies.sh [hunk 2]
```diff
@@ -3,21 +3,17 @@ set -xe
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $DIR/..
 
-which brew || /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)"
```
Comment (ianblenke) [inline Tools/dependencies.sh]: This is for stand-alone developer workstations as well as for the autobuild. If brew is in the path, it won't install. This has no speed penalty. Please leave it there.

File: Tools/dependencies.sh [hunk 3]
```diff
@@ -3,21 +3,17 @@ set -xe
...
-
-
 
 wget --no-check-certificate https://raw.githubusercontent.com/FFmpeg/gas-preprocessor/master/gas-preprocessor.pl && chmod +x gas-preprocessor.pl && sudo mv -f gas-preprocessor.pl /usr/local/bin &
 bundle install &
 
-cat .gitmodules | grep -o '".*"' | cut -d '"' -f 2 | while sleep 2 && read line; do git submodule update --init --recursive $line & done
-# make sure everything is fetched
 
-brew update 1>/dev/null && brew install doxygen nasm yasm optipng imagemagick intltool ninja antlr cmake
```
Comment (ianblenke) [inline Tools/dependencies.sh:16]: This is another list of things that I'm pretty sure a new developer station will need.

File: Tools/dependencies.sh [hunk 4]
```diff
@@ -3,21 +3,17 @@ set -xe
...
 
-cat .gitmodules | grep -o '".*"' | cut -d '"' -f 2 | while sleep 2 && read line; do git submodule update --init --recursive $line & done
-# make sure everything is fetched
 
-brew update 1>/dev/null && brew install doxygen nasm yasm optipng imagemagick intltool ninja antlr cmake 
+# make sure everything is fetched
 
-[ -x /usr/local/bin/libtoolize ] || sudo ln -sf /usr/local/bin/glibtoolize /usr/local/bin/libtoolize
 
+( which brew || /usr/bin/ruby -e "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/master/install)" ) && brew update 1>/dev/null && brew install doxygen nasm yasm optipng imagemagick intltool ninja antlr cmake &&  ([ -x /usr/local/bin/libtoolize ] || sudo ln -sf /usr/local/bin/glibtoolize /usr/local/bin/libtoolize ) &
```
Comment (ianblenke) [inline Tools/dependencies.sh:14]: If you have `set -e`, then you don't need to do this && silliness.

What is important here is the install of the other things like ninja, antlr, cmake, that are required to build. Right?

Comment (shareefalis) [inline Tools/dependencies.sh:14]: Right in order to run in a synced while having async. this allows the  chain while waiting for the in other to finish see the & in the end of the command

File: Tools/dependencies.sh (hunk 3 above)
Comment (shareefalis) [inline Tools/dependencies.sh:16]: I moved it to sync chaining step

File: Tools/dependencies.sh (hunk 1 above)
Comment (shareefalis) [inline Tools/dependencies.sh:18]: I moved it to sync chaining step

File: Tools/travis_compile.sh [hunk 5]
```diff
@@ -3,28 +3,9 @@ set -xe
...
-  echo "Could not sync s3://$CACHE/ace-ios/sdkcache/liblinphone-sdk_${SUBMODULE_HASH}_submodules.txt to sdkcache/liblinphone-sdk_${SUBMODULE_HASH}_submodules.txt"
-fi
-
-if [ -f sdkcache/liblinphone-sdk_${SUBMODULE_HASH}.zip ] ; then
-  unzip sdkcache/liblinphone-sdk_${SUBMODULE_HASH}.zip
-  exit 0
-else
-  echo "There is no existing cached SDK build for this set of submodules."
+if [!  -e liblinphone-sdk ] ; then
+  echo "Skipping building"
```
Comment (shareefalis) [inline Tools/travis_compile.sh:11]: Need fixing..  wrong logic

File: Tools/travis_depends.sh [hunk 6]
```diff
@@ -4,10 +4,10 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 cd $DIR/..
 
 mkdir -p sdkcache/
-git submodule | cut -d'(' -f1 | awk '{print substr($0, 2, length($0))}'  | sort > sdkcache/current.txt
+git submodule | cut -d'(' -f1 | awk '{print substr($0, 2, length($0))}'  | sort > current.txt
 which md5sum || brew install md5sha1sum
-SUBMODULE_HASH=$(md5sum sdkcache/current.txt | awk '{print $1}')
-mv sdkcache/current.txt liblinphone-sdk_${SUBMODULE_HASH}_submodules.txt
+SUBMODULE_HASH=$(md5sum current.txt | awk '{print $1}')
+cp current.txt liblinphone-sdk_${SUBMODULE_HASH}_submodules.txt
```
Comment (shareefalis) [inline Tools/travis_depends.sh]: not sure if we need this @ianblenke

**Comment by shareefalis** [comment]
from 30-40 minutes to 16 minutes! Wee!

File: Tools/travis_depends.sh [hunk 7]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+set -xe
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+cd $DIR/..
+git submodule update --init Classes/KIF &
+bundle install &
```
Comment (dudymas) [inline Tools/travis_depends.sh:6]: to better control these concurrent tasks, you might use a function like:

```
function spawn_for_great_justice {
local run_this=$1 wait_for=${2:-500}
bash -c “$run_this” &
local dat_pid=$!
( sleep $wait_for ; if kill -0 $dat_pid ; then kill $dat_pid ; fi ) &
}
```

and then you know that you'll have some level of control over zombies... won't work all the time, but generally safer for small builds

You'll need to change the default of wait_for as appropriate.... once you hit a timing that usually succeeds, double the number and set that in stone. If something starts exceeding that and failing, you should really consider failing the process.

**Comment by dudymas** [comment]
-1500 points for mispelling dependencies in PR title
shame

**Comment by shareefalis** [comment]
https://github.com/VTCSecureLLC/ace-ios/compare/redesigasync...experimentalwaitcommand   see here for additional improvement for zombie handling. and that wait command used for waiting for all dependencies before  going into the next script. The 2 additions goes beyond the scope of the pr  and low risk item because we will extract sdk from build server and linphone updates are not going to happen that often.   This pr looks good to me. 

Oh and @dudymas  fixed the title issues. :)

**Comment by zackm0571** [comment]
+1
